analyses/infercnv-consensus-cell-type/scripts/01_run-infercnv.R

```diff
@@ -108,22 +129,46 @@ option_list <- list(
   )
 )
 
-
-# Parse options ----------------
+# Parse and check input options ----------------
 opts <- parse_args(OptionParser(option_list = option_list))
 
-
-# check input directories
 stopifnot(
   "sce_file does not exist" = file.exists(opts$sce_file),
-  "reference_file does not exist" = file.exists(opts$reference_file),
   "gene_order_file does not exist" = file.exists(opts$gene_order_file),
   "annotation_file not provided" = !is.null(opts$annotation_file),
   "output_dir was not specified" = !is.null(opts$output_dir),
+  "reference_type was not specified" = !is.null(opts$reference_type),
   "hmm_model not properly specified" = opts$hmm_model %in% c("i3", "i6")
 )
 
 
+# check the reference information for both types of references, as different arguments are required for either type
+# also, define the ref_name while here
+stopifnot("reference_type must be one of 'pooled' or 'internal'." = opts$reference_type %in% c("pooled", "internal"))
+if (opts$reference_type == "pooled") {
+  stopifnot("pooled_reference_sce was not specified. This is required if reference_type is 'pooled'." = !is.null(opts$pooled_reference_sce))
+  stopifnot("pooled_reference_sce does not exist." = file.exists(opts$pooled_reference_sce))
```

Comment:
You could simplify this by setting the default to `""` and then just have the file exists check.